core/src/main/java/com/linecorp/armeria/server/docs/DocService.java

```diff
@@ -297,22 +310,43 @@ private CompletableFuture<AggregatedHttpFile> loadVersions(Executor executor) {
             }, executor));
         }
 
-        private CompletableFuture<AggregatedHttpFile> loadSpecifications(Executor executor) {
-            return files.computeIfAbsent(specificationPath, key -> {
-                return CompletableFuture.supplyAsync(() -> {
-                    final DocStringSupport docStringSupport = new DocStringSupport(services);
-                    ServiceSpecification spec = generate(services);
-                    spec = docStringSupport.addDocStrings(spec);
-                    spec = exampleSupport.addExamples(spec);
-                    try {
-                        final byte[] content = jsonMapper.writerWithDefaultPrettyPrinter()
-                                                         .writeValueAsBytes(spec);
-                        return toFile(content, MediaType.JSON_UTF_8);
-                    } catch (JsonProcessingException e) {
-                        throw new RuntimeException(e);
-                    }
-                }, executor);
-            });
+        private CompletableFuture<ServiceSpecification> generateServiceSpecification(Executor executor) {
+            return CompletableFuture.supplyAsync(() -> {
+                final DocStringSupport docStringSupport = new DocStringSupport(services);
+                ServiceSpecification spec = generate(services);
+                spec = docStringSupport.addDocStrings(spec);
+                spec = exampleSupport.addExamples(spec);
+                return spec;
+            }, executor);
+        }
+
+        private CompletableFuture<AggregatedHttpFile> loadSpecifications(
+                CompletableFuture<ServiceSpecification> specificationFuture) {
+            return files.computeIfAbsent(SPECIFICATION_PATH, key -> specificationFuture.thenApply((spec) -> {
+                try {
+                    final byte[] content = jsonMapper.writerWithDefaultPrettyPrinter()
+                                                     .writeValueAsBytes(spec);
+                    return toFile(content, MediaType.JSON_UTF_8);
+                } catch (JsonProcessingException e) {
+                    throw new RuntimeException(e);
+                }
+            }));
+        }
+
+        private CompletableFuture<AggregatedHttpFile> loadSchemas(
+                CompletableFuture<ServiceSpecification> specificationFuture) {
+            return files.computeIfAbsent(SCHEMAS_PATH, key -> specificationFuture.thenApply((spec) -> {
+                try {
+                    final ArrayNode jsonSpec = JsonSchemaGenerator.generate(spec);
+
+                    final byte[] content = jsonMapper.writerWithDefaultPrettyPrinter()
+                                                     .writeValueAsBytes(jsonSpec);
+                    return toFile(content, MediaType.JSON_UTF_8);
+                } catch (JsonProcessingException e) {
```

Comment:
Sorry 😅    This isn't related to your PR, but do you mind adding the following logs to L201?

https://github.com/line/armeria/blob/afcd0ba659f190b44c31a2f70f13553ef9bc1f97/core/src/main/java/com/linecorp/armeria/server/docs/DocService.java#L200-L202

The main concern is an exception is raised somewhere, but we don't know what exception was actually thrown. (if it's not a `JsonProcessingException`)

```
if (e != null) {
    logger.warn("Failed to load specifications completely: ", e);
}
```

Reply:
Done 👍